toolkit/tools/specreader/specreader.go

```diff
@@ -26,78 +30,192 @@ const (
 	defaultWorkerCount = "10"
 )
 
+// parseResult holds the worker results from parsing a SPEC file.
+type parseResult struct {
+	packages []*pkgjson.Package
+	err      error
+}
+
 var (
-	app      = kingpin.New("specreader", "A tool to parse spec dependencies into JSON")
-	dir      = exe.InputDirFlag(app, "Directory to scan for SPECS")
-	output   = exe.OutputFlag(app, "Output file to export the JSON")
-	workers  = app.Flag("workers", "Number of concurrent goroutines to parse with").Default(defaultWorkerCount).Int()
-	srpmDir  = app.Flag("srpm-dir", "Directory containing SRPMs.").Required().ExistingDir()
-	macroDir = app.Flag("macro-dir", "Directory containing rpm macros.").Default("").String()
-	distTag  = app.Flag("dist-tag", "The distribution tag the SPEC will be built with.").Required().String()
-	logFile  = exe.LogFileFlag(app)
-	logLevel = exe.LogLevelFlag(app)
+	app       = kingpin.New("specreader", "A tool to parse spec dependencies into JSON")
+	specsDir  = exe.InputDirFlag(app, "Directory to scan for SPECS")
+	output    = exe.OutputFlag(app, "Output file to export the JSON")
+	workers   = app.Flag("workers", "Number of concurrent goroutines to parse with").Default(defaultWorkerCount).Int()
+	buildDir  = app.Flag("build-dir", "Directory to store temporary files while parsing.").String()
+	srpmsDir  = app.Flag("srpm-dir", "Directory containing SRPMs.").Required().ExistingDir()
+	rpmsDir   = app.Flag("rpm-dir", "Directory containing built RPMs.").Required().ExistingDir()
+	distTag   = app.Flag("dist-tag", "The distribution tag the SPEC will be built with.").Required().String()
+	workerTar = app.Flag("worker-tar", "Full path to worker_chroot.tar.gz").ExistingFile()
+	logFile   = exe.LogFileFlag(app)
+	logLevel  = exe.LogLevelFlag(app)
 )
 
 func main() {
 	app.Version(exe.ToolkitVersion)
 	kingpin.MustParse(app.Parse(os.Args[1:]))
 	logger.InitBestEffort(*logFile, *logLevel)
 
+	if *workers <= 0 {
+		logger.Log.Panicf("Value in --workers must be greater than zero. Found %d", *workers)
+	}
+
+	err := parseSPECsWrapper(*buildDir, *specsDir, *rpmsDir, *srpmsDir, *distTag, *output, *workerTar, *workers)
+	logger.PanicOnError(err)
+}
+
+// parseSPECsWrapper wraps parseSPECs to conditionally run it inside a chroot.
+// If workerTar is non-empty, parsing will occur inside a chroot, otherwise it will run on the host system.
+func parseSPECsWrapper(buildDir, specsDir, rpmsDir, srpmsDir, distTag, outputFile, workerTar string, workers int) (err error) {
+	var (
+		chroot      *safechroot.Chroot
+		packageRepo *pkgjson.PackageRepo
+	)
+
+	if workerTar != "" {
+		const leaveFilesOnDisk = false
+		chroot, err = createChroot(workerTar, buildDir, specsDir, srpmsDir)
+		if err != nil {
+			return
+		}
+		defer chroot.Close(leaveFilesOnDisk)
+	}
+
+	doParse := func() error {
+		var err error
```

Comment:
>var err error [](start = 2, length = 13)

Why not just use the `err` we already have in the outer function? Also, I'd use a different name just to be clear which variable is being assigned a value.

Reply:
This routine is given to `chroot.Run` which expects a routine that returns an error object. I'd rather not side step that flow incase in the future we decide to make safechroot perform additional actions when a `Run` returns an error. 

I'll clarify the variable names.